ui-speedspacechart/src/components/common/DetailsBox.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { MARGINS } from '../const';
+
+type DetailsBoxProps = {
+  width: number;
+  height: number;
+  curveX: number;
+  curveY: number;
+  marecoSpeedText: string;
+  effortText: string;
+  electricalProfileText: string;
+  previousGradientText: number;
+  modeText: string;
+};
+
+const DetailsBox = ({
+  width,
+  height,
+  curveX,
+  curveY,
+  marecoSpeedText,
+  effortText,
+  electricalProfileText,
+  previousGradientText,
+  modeText,
+}: DetailsBoxProps) => {
+  const { MARGIN_TOP, MARGIN_BOTTOM, MARGIN_LEFT, MARGIN_RIGHT } = MARGINS;
+
+  let rightOffset = 0;
+  let bottomOffset = 0;
+
+  if (curveX + MARGIN_LEFT + 115 > width - MARGIN_RIGHT - 10) {
```

Comment:
what does this condition mean, might need some comments to help future readers

Reply:
commented